### PR TITLE
Fabric RL example: add `torchmetrics` and minor fixes

### PR DESCRIPTION
**Pull request by belerico**
## What does this PR do?

In this PR i've added the following:

* Reward and episode length per player (on rank-0) are substituted by the average reward and average episode length computed as the average over all the `--per-rank-num-envs` in the world. For the Fabric example i've used the `torchmetrics` package, while in the PyTorch one i've just collected stats using the standard `torch.distributed` collective functions. The user should appreciate the flexibility of `torchmetrics`
* Improved value loss
* Logs folder are now organized to contain both `fabric_logs` and `torch_logs` for better comparison
* `--capture-video` flag now creates folder for train and test videos separately
* For the PyTorch example, a `--cuda` flag specifies whether to run training over GPU (if those are available)

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: examples/fabric/reinforcement_learning/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
-gymnasium
+gymnasium[box2d]>=0.27.1
```
Comment (Borda) [inline examples/fabric/reinforcement_learning/requirements.txt:1]: Not sure if we apply these requirements... :/

**Review by Borda — APPROVED** [review]

File: examples/fabric/reinforcement_learning/requirements.txt (hunk 1 above)
Comment (awaelchli) [reply examples/fabric/reinforcement_learning/requirements.txt:1]: They are for documentation, for users who want to try the example

**Review by awaelchli — APPROVED** [review]
